Messaging.md

```diff
@@ -363,92 +371,210 @@ Witnesses should be sorted by the `input_serial_id` sent in `funding_input` defi
 
 This type contains information provided by an oracle on an event that it will attest to.
 See [the Oracle specifications](./Oracle.md#oracle-event) for more details.
 
-**For backward compatibility reasons, this type is currently serialized as a TLV and uses u16 for collection prefix. It is expected to be changed to the new serialization format in a near future.**
-
 #### `oracle_event`
 
-1. type: 55330
 1. data:
-   * [`u16`:`nb_nonces`]
-   * [`nb_nonces*x_point`:`oracle_nonces`]
-   * [`u32`:`event_maturity_epoch`]
+   * [`oracle_event_timestamp`:`timestamp`]
    * [`event_descriptor`:`event_descriptor`]
    * [`string`:`event_id`]
 
+### The `oracle_metadata` Type
+
+This type contains static oracle information and can be used to import trusted oracles into wallets.
+This information should be saved by DLC nodes.
+
+#### `oracle_metadata`
+
+1. data:
+   * [`x_point`:`announcement_public_key`]
+   * [`string`:`oracle_name`]
+   * [`string`:`oracle_description`]
+   * [`u32`:`timestamp`]
+   * [`oracle_schemes`:`oracle_schemes`]
+   * [`signature`:`oracle_metadata_signature`]
+
+where the `oracle_metadata_signature` is a Schnorr signature by the `announcement_public_key` using the tag `oraclemetadata/v0` of the sha256 hash of the serialized `oracle_metadata` message omitting the `announcement_public_key` and of course the the `oracle_metadata_signature`.
+
+#### Rationale
+
+Separate public keys are required for announcement and attestation because the attestation scheme
+used in the current DLC spec reduces the security of non-attestation Schnorr signatures issued by the
+attestation keys and hence a separate announcement key is required.
+
+The `oracle_metadata_signature` is used to link all data in this message to the `announcement_public_key`.
+
+#### Requirements
+
+A node
+
+* SHOULD save this information in persistent storage.
+* MUST reject if `announcement_public_key == attestation_public_key`.
+
+
+### The `proof_of_knowledge` Type
+
+A proof of knowledge is used to prove that oracles know the pre-images of the nonces they are using to prevent cancellation attacks in multi-oracle settings.
+
+#### `schnorr_proof_of_knowledge`
+
+1. implements: `proof_of_knowledge`
+1. type: 0
+1. data:
+   * [`signature`:`attestation_public_key_proof`]
+   * [`nb_signatures`:`bigsize`]
+   * [`nb_signatures*signature`:`nonce_proofs`]
+
+where each signature is generated over the message `dlcoraclepok`.
+
+### The `oracle_schemes` Type
+
+In order to enable oracles to support multiple attestation schemes, and clients to chose their preferred one, the `oracle_schemes` type can contain multiple attestations schemes.
+Each scheme is defined as a TLV so that clients can easily ignore schemes that they don't know about.
+Note that at the moment the only defined scheme is the `schnorr_scheme` that uses Schnorr signatures.
+
+#### `oracle_schemes`
+
+1. data:
+   * [`bigsize`:`nb_schemes`]
+   * [`nb_schemes*scheme`:`attestation_schemes`]
```

Comment:
Is `attestations_schemes` defined anywhere or is this a typo and meant to be `schnorr_scheme`?

Reply:
Added a `attestation_scheme` type.